pages/index.tsx

```diff
@@ -30,63 +30,76 @@ interface HomeProps {
 
 function Home(props: HomeProps) {
     const LIMIT = 96;
-    const isBrowser = typeof window !== `undefined`
+    const isBrowser = typeof window !== `undefined`;
     const client = useApolloClient();
 
     const [state, setState] = useState<InititalState>(INITITAL_STATE);
-    const [initLoading , setInitLoading] = useState<boolean>(true);
+    const [initLoading, setInitLoading] = useState<boolean>(true);
     const [loadingExtra, setLoadingExtra] = useState<boolean>(true);
     const [isAtBottomOfPage, setIsAtBottomOfPage] = useState(false);
 
     useEffect(function initializeView() {
         if (isBrowser) {
-            window.addEventListener('scroll', checkIsBottomPage)
-            return () => window.removeEventListener('scroll', checkIsBottomPage);
+            window.addEventListener('scroll', checkIsBottomPage);
+            return () =>
+                window.removeEventListener('scroll', checkIsBottomPage);
         }
-    }, [])
-
-    useEffect(function handleTabChange() {
-        fetchMoreWhenSearched()
-    }, [state.activeTab])
-
-    useEffect(function handleRefetchingOnBottomOfPage() {
-        const isEndReached = state.keycapsets.length === state.allKeycapsetsCount;
-
-        if (isEndReached) {
-            setLoadingExtra(false);
-            return;
-        }
-        if (isAtBottomOfPage) {
-            setLoadingExtra(true);
-            fetchMoreWhenBottomOfPage();
-            setIsAtBottomOfPage(false);
-            return;
-        }
-    }, [isAtBottomOfPage])
-
-
-    useEffect(function handleSearch() {
-        let timeout: any;
-        clearTimeout(timeout);
-
-        timeout = setTimeout(() => {
-            if(state.searchQuery !== '' || state.searchQuery !== undefined) {
-                fetchMoreWhenSearched();
-                forceCheck();
-            } else {
-                initSets();
+    }, []);
+
+    useEffect(
+        function handleTabChange() {
+            fetchMoreWhenSearched();
+        },
+        [state.filters.activeTab, state.filters.availabilityFilter]
+    );
+
+    useEffect(
+        function handleRefetchingOnBottomOfPage() {
+            const isEndReached =
+                state.keycapsets.length === state.allKeycapsetsCount;
```

Comment:
The formatting here was also fine how it was? this is adding more lines and more unreadability for now reason imo.

Reply:
How is the prettiest integration going? Running that in this PR would resolve quite a few issues you’ve mentioned.